### PR TITLE
[test] [bug-fix] Many openssl s_client tests fail occasionally due to timing issues

### PR DESCRIPTION
**Pull request by nalramli**
## Overview
There are many tests that utilize `openssl s_client` which are flaky.  They sometimes fail randomly with an error that includes the tool’s output.  The failures are usually timing issues which are not predictable and subsequent attempts to run the tests often succeed.

This patch unifies the way that the tool is called across all unit tests using the `run_openssl_client()` method to get a more consistent behavior.  The method has been enhanced to use `open3()`, then `waitpid()` until the process ends with a configurable timeout, optionally insert data using `stdin`, and capture `stdout`  and `stderr` separately.

### CODE REVIEW

File: t/Util.pm
Comment (kazuho) [inline t/Util.pm:835]:
```suggestion
    my $request_default = $opts->{request_default} // undef;
```
Assuming that this is a boolean, the custom is to represent false using an undef.

File: t/Util.pm
Comment (kazuho) [inline t/Util.pm:839]: Are we using tempdir at all? If not, we can remove this. Otherwise, it might be a good idea to set `CLEANUP => 1` so that the temporary directory would be removed when the object is discarded.

File: t/Util.pm
Comment (kazuho) [inline t/Util.pm:845]: Just a remainder that IPC::Open3 is sensitive to deadlocks. As it uses pipe in both directions, we might end up seeing the sub-command blocked on write, while us being blocked on write. I would assume that the code below would block for example when the server is a simple echo server and the we write lots of data.

The way to address this problem is to redirect stdout and stderr of the sub-command to files in temporary directory, then read them once the command execution completes.

I would not mind using IPC::Open3 for the time being, but pointing that out as I see tempdir above.

File: t/Util.pm
Comment (kazuho) [inline t/Util.pm:846]:
```suggestion
    sleep $timeout;
```

File: t/Util.pm
Comment (kazuho) [inline t/Util.pm:849]:
```suggestion
```
I do not think we need to set autoflush for read-only handles?

File: t/Util.pm
Comment (kazuho) [inline t/Util.pm:886]: Do we need to define both `run_openssl_client` and `run_openssl_client_joined`? I ask this because I do not see former being called. Also, "joined" sounds a bit odd to me - it does not parse well unless to those who understand what the function does.

Would it make sense to return the joined string by default from `run_openssl_client`, while (potentially) adding a option to return an array instead?

**Review by kazuho — COMMENTED** [review]
Thank you for the nice cleanup! The PR looks good to go modulo minor concerns.

File: t/Util.pm
Comment (nalramli) [reply t/Util.pm:844]: I agree.  Since we don't have such test scenarios, let's keep it as is?

File: t/Util.pm
Comment (nalramli) [reply t/Util.pm]: You're right.  We are not.  This has been removed in https://github.com/h2o/h2o/pull/2947/commits/507bd47dd6c7174525e77f698af658d4fe47d287.

File: t/Util.pm
Comment (nalramli) [reply t/Util.pm:885]: This has been incorporated in https://github.com/h2o/h2o/pull/2947/commits/ff454017b1adf6f9eab8d7caf7c5015df786db68.  I wasn't sure what to name the option, I used `split_return`, please let me know if you think I should change it.

**Review by kazuho — APPROVED** [review]
Thank you for the changes.

**Pull request merged**
